cmd/context.go

```diff
@@ -0,0 +1,239 @@
+		Short:  "List contexts",
+		Use:    "list <vcs-type> <org-name>",
+		PreRun: initClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return listContexts(cl, args[0], args[1])
+		},
+		Args: cobra.ExactArgs(2),
+	}
+
+	showContextCommand := &cobra.Command{
+		Short:  "Show a context",
+		Use:    "show <vcs-type> <org-name> <context-name>",
+		PreRun: initClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return showContext(cl, args[0], args[1], args[2])
+		},
+		Args: cobra.ExactArgs(3),
+	}
+
+	storeCommand := &cobra.Command{
+		Short:  "Store an new secret in the named context. The value is read from stdin.",
+		Use:    "store-secret <vcs-type> <org-name> <context-name> <secret name>",
+		PreRun: initClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return storeEnvVar(cl, args[0], args[1], args[2], args[3])
+		},
+		Args: cobra.ExactArgs(4),
+	}
+
+	removeCommand := &cobra.Command{
+		Short:  "Remove a secret from the named context",
+		Use:    "remove-secret <vcs-type> <org-name> <context-name> <secret name>",
+		PreRun: initClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return removeEnvVar(cl, args[0], args[1], args[2], args[3])
+		},
+		Args: cobra.ExactArgs(4),
+	}
+
+	createContextCommand := &cobra.Command{
+		Short:  "Create a new context",
+		Use:    "create <vcs-type> <org-name> <context-name>",
+		PreRun: initClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createContext(cl, args[0], args[1], args[2])
+		},
+		Args: cobra.ExactArgs(3),
+	}
+
+	force := false
+	deleteContextCommand := &cobra.Command{
+		Short:  "Delete the named context",
+		Use:    "delete <vcs-type> <org-name> <context-name>",
+		PreRun: initClient,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deleteContext(cl, force, args[0], args[1], args[2])
+		},
+		Args: cobra.ExactArgs(3),
+	}
+
+	deleteContextCommand.Flags().BoolVarP(&force, "force", "f", false, "Delete the context without asking for confirmation.")
+
+	command.AddCommand(listCommand)
+	command.AddCommand(showContextCommand)
+	command.AddCommand(storeCommand)
+	command.AddCommand(removeCommand)
+	command.AddCommand(createContextCommand)
+	command.AddCommand(deleteContextCommand)
+
+	return command
+}
+
+func listContexts(client *client.Client, vcs, org string) error {
+
+	contexts, err := api.ListContexts(client, org, vcs)
+
+	if err != nil {
+		return err
+
+	}
+
+	table := tablewriter.NewWriter(os.Stdout)
+
+	table.SetHeader([]string{"Provider", "Organization", "Name", "Created At"})
+
+	for _, context := range contexts.Organization.Contexts.Edges {
+
+		table.Append([]string{
+			vcs,
+			org,
+			context.Node.Name,
+			context.Node.CreatedAt,
+		})
+	}
+	table.Render()
+
+	return nil
+}
+
+func contextByName(client *client.Client, vcsType, orgName, contextName string) (*api.CircleCIContext, error) {
+
+	contexts, err := api.ListContexts(client, orgName, vcsType)
```

Comment:
Can the graphQL api filter for us, rather than having to download the whole list including env vars every time?